lib/src/ui/pages/home/home_page.dart

```diff
@@ -1,248 +1,38 @@
-        ),
-      ),
-      backgroundColor: Theme.of(context).backgroundColor,
-      actions: [
-        Padding(
-          padding: const EdgeInsets.symmetric(
-            vertical: _Constants.appBarActionsVerticalPadding,
-          ),
-          child: PrimaryIconButton(
-            iconData: Icons.search,
-            onPressed: () {},
-          ),
-        ),
-        Padding(
-          padding: const EdgeInsets.only(
-            top: _Constants.appBarActionsVerticalPadding,
-            bottom: _Constants.appBarActionsVerticalPadding,
-            right: Dimens.pagePadding,
-            left: _Constants.appBarActionsIconsPadding,
-          ),
-          child: PrimaryIconButton(
-            iconData: Icons.settings,
-            onPressed: () {
-              Navigator.push(
-                context,
-                MaterialPageRoute(
-                  builder: (context) => ChangeNotifierProvider(
-                    create: (_) => inject<SettingsNotifier>(),
-                    child: SettingsPage(),
-                  ),
-                ),
-              );
-            },
-          ),
-        ),
-      ],
-    );
-  }
-
-  Widget _buildNewsSectionHeader({
-    @required String title,
-    @required WidgetBuilder routeBuilder,
-  }) {
-    return Padding(
-      padding: const EdgeInsets.symmetric(horizontal: Dimens.pagePadding),
-      child: Row(
-        mainAxisAlignment: MainAxisAlignment.spaceBetween,
-        children: [
-          Flexible(
-            child: Hero(
-              tag: title,
-              child: Text(
-                title,
-                style: Theme.of(context).textTheme.headline3,
-              ),
-            ),
-          ),
-          PrimaryTextButton(
-            text: Strings.of(context).showAll,
-            onPressed: () => Navigator.push(
-              context,
-              MaterialPageRoute(
-                builder: routeBuilder,
-              ),
-            ),
-          ),
-        ],
-      ),
-    );
-  }
-
-  Widget _buildSecondaryNewsItems(List<Article> articles) => Column(
-        children: articles
-            .toSecondaryNewsListEntities()
-            .take(3)
-            .toList()
-            .map((news) => SecondaryNewsListItem(news: news))
-            .toList(),
-      );
-
-  Widget _errorContent(BuildContext providerContext, HomeNotifierErrorState state) => Center(
-    //TODO use state to display error returned from the notifier
-        child: Container(
-          height: _Constants.retryButtonHeight,
-          width: _Constants.retryButtonWidth,
-          child: PrimaryTextButton(
-            onPressed: () => providerContext.read<HomeNotifier>().fetchHomeArticles(),
-            text: Strings.of(context).retry,
-          ),
+  Widget build(BuildContext context) => Scaffold(
+        body: _pages[_selectedPage],
+        bottomNavigationBar: BottomNavigationBar(
+          onTap: (index) => setState(() => _selectedPage = index),
```

Comment:
wiele apek ma coś takiego, że przy kliknięciu wstecz wraca do `startowej zakładki` i dopiero z niej poprzez back button wychodzi się z apsy i tu w sumie proponowałbym to samo, tj. jak kliknie wstecz na profilu to żeby przeniosło do FeedPage :)